src/playground/services/session/login/endpoint.clj

```diff
@@ -0,0 +1,38 @@
+(ns playground.services.session.login.endpoint
+  (:require [clojure.java.jdbc :as jdbc]
+            [honeysql.core :as h]
+            [clojure.spec.alpha :as spec]
+            [ring.util.response :as ring-resp]
+            [playground.services.session.login.logic :as logic]))
+
+(spec/def ::username (spec/and string? (spec/nilable not-empty)))
```

Comment:
`(spec/nilable not-empty)` doesn't make sense.

Maybe you meant `(spec/and string? seq)`

Reply:
Testing now it gives the same spec -> excluding a blank . I'll change to `(spec/and string? seq)`